core/drand_beacon.go

```diff
@@ -0,0 +1,348 @@
+package core
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/drand/drand/common"
+
+	"github.com/drand/drand/chain"
+	"github.com/drand/drand/chain/boltdb"
+	"github.com/drand/drand/fs"
+
+	"github.com/drand/drand/chain/beacon"
+	"github.com/drand/drand/key"
+	dlog "github.com/drand/drand/log"
+	"github.com/drand/drand/net"
+	"github.com/drand/kyber/share/dkg"
+)
+
+// BeaconProcess is the main logic of the program. It reads the keys / group file, it
+// can start the DKG, read/write shars to files and can initiate/respond to TBlS
+// signature requests.
+type BeaconProcess struct {
+	opts *Config
+	priv *key.Pair
+	// current group this drand node is using
+	group *key.Group
+	index int
+
+	store       key.Store
+	privGateway *net.PrivateGateway
+	pubGateway  *net.PublicGateway
+
+	beacon *beacon.Handler
+
+	// dkg private share. can be nil if dkg not finished yet.
+	share   *key.Share
+	dkgDone bool
+
+	// manager is created and destroyed during a setup phase
+	manager  *setupManager
+	receiver *setupReceiver
+
+	// dkgInfo contains all the information related to an upcoming or in
+	// progress dkg protocol. It is nil for the rest of the time.
+	dkgInfo *dkgInfo
+	// general logger
+	log dlog.Logger
+
+	// global state lock
+	state  sync.Mutex
+	exitCh chan bool
+
+	// that cancel function is set when the drand process is following a chain
+	// but not participating. Drand calls the cancel func when the node
+	// participates to a resharing.
+	syncerCancel context.CancelFunc
+
+	// only used for testing currently
+	// XXX need boundaries between gRPC and control plane such that we can give
+	// a list of paramteres at each DKG (inluding this callback)
+	setupCB func(*key.Group)
+
+	// version indicates the base code variant
+	version common.Version
+
+	// only used for testing at the moment - may be useful later
+	// to pinpoint the exact messages from all nodes during dkg
+	dkgBoardSetup func(Broadcast) Broadcast
+}
+
+func NewBeaconProcess(log dlog.Logger, version common.Version, store key.Store,
```

Comment:
I think we dont need to pass version since this is fixed by the binary anyway
https://github.com/drand/drand/pull/846/files#diff-95f9dbb62f1d2e017822b23a6f56f2b4844a937003c0199f569993925eebd027R51 - every places that need the version should use `common.GetAppVersion()`.

Reply:
ok, fixed!